src/EnergyPlus/PlantHeatExchangerFluidToFluid.cc

```diff
@@ -1998,4 +2001,101 @@ void HeatExchangerStruct::oneTimeInit(EnergyPlusData &state)
+    auto &supplyCoilData = state.dataPlnt->PlantLoop(this->SupplySideLoop.loopNum).compDesWaterFlowRate;
+    std::vector<Real64> supplyFlowData;
+    supplyFlowData.resize(size_t(24 * state.dataGlobal->TimeStepsInHour + 1));
+    for (size_t i = 0; i < supplyFlowData.size(); ++i) {
+        supplyFlowData[i] = 0.0;
+    }
+    if (supplyCompSize > 0) {
+        for (size_t comp = 0; comp < supplyCoilData.size(); ++comp) {
+            if (comp == supplyArrayIndex) {
+                continue;
+            }
+            for (size_t ts = 0; ts < supplyCoilData[comp].tsDesWaterFlowRate.size(); ++ts) {
+                supplyFlowData[ts] += supplyCoilData[comp].tsDesWaterFlowRate[ts];
+            }
+            supplyFlowData[0] = -1;
+        }
+        if (demandArrayIndex == -1) {
+            demandCompNames.emplace_back(this->Name);
+            demandCompTypes.emplace_back(this->SupplySideLoop.comp->Type);
+            size_t arrayIndex = demandCoilData.size() + 1;
+            demandCoilData.resize(arrayIndex);
+            demandCoilData[arrayIndex - 1].tsDesWaterFlowRate.resize(size_t(24 * state.dataGlobal->TimeStepsInHour));
+            demandCoilData[arrayIndex - 1].tsDesWaterFlowRate = supplyFlowData;
+        } else {
+            demandCoilData[demandArrayIndex].tsDesWaterFlowRate = supplyFlowData;
+        }
+    }
+    if (hasSupplySideTES(state)) {
+        // if the supply side of the HX contains a TES system then copy demand side coil data to supply side so TES can size on the whole load
+        std::vector<Real64> demandFlowData;
+        demandFlowData.resize(size_t(24 * state.dataGlobal->TimeStepsInHour + 1));
+        for (size_t i = 0; i < demandFlowData.size(); ++i) {
+            demandFlowData[i] = 0.0;
+        }
+        if (demandCompSize > 0) {
+            for (size_t comp = 0; comp < demandCoilData.size(); ++comp) {
+                if (comp == demandArrayIndex) {
+                    continue;
+                }
+                for (size_t ts = 0; ts < demandCoilData.size(); ++ts) {
+                    demandFlowData[ts] += demandCoilData[comp].tsDesWaterFlowRate[ts];
+                }
+                demandFlowData[0] = -1;
+            }
+        }
+        if (demandCompSize > 0) {
+            if (supplyArrayIndex == -1) {
+                supplyCompNames.emplace_back(this->Name);
+                supplyCompTypes.emplace_back(this->SupplySideLoop.comp->Type);
+                size_t arrayIndex = supplyCoilData.size() + 1;
+                supplyCoilData.resize(arrayIndex);
+                supplyCoilData[arrayIndex - 1].tsDesWaterFlowRate.resize(size_t(24 * state.dataGlobal->TimeStepsInHour));
+                supplyCoilData[arrayIndex - 1].tsDesWaterFlowRate = demandFlowData;
+            } else {
+                supplyCoilData[demandArrayIndex].tsDesWaterFlowRate = demandFlowData;
+            }
+        }
+    }
+}
+
+bool HeatExchangerStruct::hasSupplySideTES(EnergyPlusData &state)
```

Comment:
Check for supply side TES. Data moves upstream unless TES is downstream and needs that data.